core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -267,6 +310,96 @@ object UnitAutomation {
         unit.movement.headTowards(settlerOrGreatPersonToAccompany.currentTile)
         return true
     }
+    
+    private fun tryPatrolBoarders(unit: MapUnit): Boolean {
+        if (unit.type == UnitType.Scout) return false
+
+        val friendlyCitiesTiles = unit.civInfo.cities
```

Comment:
.sequence() this first so the rest is sequences, also the city has a helper function to help you get the actual tiles